lib/ide/parser/src/api.rs

```diff
@@ -15,9 +59,14 @@ pub use ast::Ast;
 pub trait IsParser : Debug {
     /// Parse program.
     fn parse(&mut self, program:String, ids:IdMap) -> Result<Ast>;
+
+    /// Parse a module content that contains idmap and metadata.
```

Comment:
We should be careful with words, when we didn't moved metadata into the module.

Module is the AST node. It has no metadata.

I'd write this more like:
`Parse contents of the program source file, where program code may be followed by idmap and metadata.`